packages/init/src/SliceMachineInitProcess.ts

```diff
@@ -0,0 +1,1114 @@
+		assertExists(
+			this.context.userRepositories,
+			"User repositories must be available through context to run `trySelectExistingRepository`",
+		);
+
+		const { maybeDomain } = await prompt<string, "maybeDomain">({
+			type: "select",
+			name: "maybeDomain",
+			message: "Pick a repository to connect to or choose to create a new one",
+			warn: "You are not a developer or admin of this repository",
+			choices: [
+				{
+					title: "CREATE NEW",
+					description: "Create a new Prismic repository\n",
+					value: "",
+				},
+				...this.context.userRepositories
+					.map((repository) => {
+						const hasWriteAccess =
+							this.manager.prismicRepository.hasWriteAccess(repository);
+
+						return {
+							title: `${repository.domain}${
+								hasWriteAccess ? "" : " (Unauthorized)"
+							}`,
+							description: `Connect to ${chalk.cyan(repository.domain)}`,
+							value: repository.domain,
+							disabled: !hasWriteAccess,
+						};
+					})
+					.sort((a, b) => (a.value > b.value ? 1 : -1)),
+			],
+		});
+
+		if (maybeDomain) {
+			this.context.repository = {
+				domain: maybeDomain,
+				exists: true,
+			};
+		}
+	}
+
+	protected async selectNewRepository(): Promise<void> {
+		let suggestedName = "";
+
+		// TODO: Improve concurrency
+		const trySuggestName = async (name: string): Promise<string | void> => {
+			if (name) {
+				const formattedName = formatRepositoryDomain(name);
+
+				if (
+					!validateRepositoryDomain({ domain: formattedName }).hasErrors &&
+					!(await this.manager.prismicRepository.checkExists({
+						domain: formattedName,
+					}))
+				) {
+					return formattedName;
+				}
+			}
+		};
+
+		// 1. Try to suggest name after package name
+		try {
+			const pkgJSONPath = path.join(this.manager.cwd, "package.json");
+			const pkg = JSON.parse(await fs.readFile(pkgJSONPath, "utf-8"));
+
+			const maybeSuggestion = await trySuggestName(pkg.name);
+			if (maybeSuggestion) {
+				suggestedName = maybeSuggestion;
+			}
+		} catch {
+			// Noop
+		}
+
+		// 2. Try to suggest name after directory name
+		if (!suggestedName) {
+			const maybeSuggestion = await trySuggestName(
+				path.basename(this.manager.cwd),
+			);
+			if (maybeSuggestion) {
+				suggestedName = maybeSuggestion;
+			}
+		}
+
+		// 3. Use random name
+		if (!suggestedName) {
+			do {
+				suggestedName = getRandomRepositoryDomain();
+			} while (
+				await this.manager.prismicRepository.checkExists({
+					domain: suggestedName,
+				})
+			);
+		}
+
+		const { domain } = await prompt<string, "domain">({
+			type: "text",
+			name: "domain",
+			// Overriden by the `onRender` function, just used as a fallback
+			message: "Choose a name for your Prismic repository",
+			initial: suggestedName,
+			onRender() {
+				const rawDomain = this.value || this.initial || "";
+				const domain = formatRepositoryDomain(rawDomain);
+				const validation = validateRepositoryDomain({ domain });
+
+				const minRule = validation.LessThan4
```

Comment:
Unnecessary

Reply:
4 character minimum is enforced in the UI and in the API:
<img width="625" alt="image" src="https://user-images.githubusercontent.com/8601064/221691208-9009f927-a6dd-4a6b-8dd6-5e38f2bc17e0.png">